frontend/src/metabase/common/components/FilterPicker/CoordinateFilterPicker/CoordinateFilterPicker.tsx

```diff
@@ -0,0 +1,307 @@
+            autoFocus
+          />
+          <Text mx="sm">{t`and`}</Text>
+          <NumberInput
+            value={values[1]}
+            onChange={(newValue: number) => onChange([values[0], newValue])}
+            placeholder={placeholder}
+          />
+        </Flex>
+      );
+    case 4:
+      return (
+        <Stack align="center" justify="center" spacing="sm" p="md">
+          <NumberInput
+            value={values[0]}
+            onChange={(newValue: number) =>
+              onChange([newValue, values[1], values[2], values[3]])
+            }
+            placeholder={insideLabels[0]}
+            autoFocus
+          />
+          <Flex align="center" justify="center" gap="sm">
+            <NumberInput
+              value={values[1]}
+              onChange={(newValue: number) =>
+                onChange([values[0], newValue, values[2], values[3]])
+              }
+              placeholder={insideLabels[1]}
+            />
+            <NumberInput
+              value={values[3]}
+              onChange={(newValue: number) =>
+                onChange([values[0], values[1], values[2], newValue])
+              }
+              placeholder={insideLabels[2]}
+            />
+          </Flex>
+          <NumberInput
+            value={values[2]}
+            onChange={(newValue: number) =>
+              onChange([values[0], values[1], newValue, values[3]])
+            }
+            placeholder={insideLabels[3]}
+          />
+        </Stack>
+      );
+    default:
+      return null;
+  }
+}
+
+function CoordinateColumnSelect({
+  query,
+  stageIndex,
+  column,
+  value,
+  onChange,
+}: {
+  query: Lib.Query;
+  stageIndex: number;
+  column: Lib.ColumnMetadata;
+  value: Lib.ColumnMetadata | null;
+  onChange: (column: Lib.ColumnMetadata) => void;
+}) {
+  const latitudeColumns = findLatitudeColumns(query, stageIndex);
+  const longitudeColumns = findLongitudeColumns(query, stageIndex);
+
+  const columnDirection = Lib.isLatitude(column) ? "latitude" : "longitude";
+
+  if (columnDirection === "latitude" && longitudeColumns.length === 1) {
+    return null;
+  }
+
+  if (columnDirection === "longitude" && latitudeColumns.length === 1) {
+    return null;
+  }
+
+  const selectLabel =
+    columnDirection === "latitude"
+      ? t`Select longitude column`
+      : t`Select latitude column`;
+
+  const options = (
+    columnDirection === "latitude"
+      ? () => getColumnOptions({ query, stageIndex, columns: longitudeColumns })
+      : () => getColumnOptions({ query, stageIndex, columns: latitudeColumns })
+  )();
+
+  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedOption = options.find(
+      option => option.value === e.target.value,
+    );
+
+    if (selectedOption) {
+      onChange(selectedOption.column);
+    }
+  };
+
+  return (
+    <Stack p="md" spacing="sm">
+      <label>
+        <strong>{selectLabel}</strong>
+      </label>
+      <Select
```

Comment:
Please use `Select` from mantine. You would need to pass `withinPortal={false}` to make it work with our deprecated Popover

Reply:
ah, maybe that's what I was missing, because it was still crashing for me